Assignment4/NumberSet/Writer/SetWriter.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NumberSet.Writer
+{
+    public class SetWriter : IDisposable
+    {
+
+        private StreamWriter? streamWriter;

```

Comment:
```suggestion
        private StreamWriter _StreamWriter;
```
You don't need to make this nullable as it will always be set when you instantiate this object.